src/components/visualizations/chart/highcharts/getAxisNameConfiguration.ts

```diff
@@ -0,0 +1,54 @@
+// (C) 2019 GoodData Corporation
+import get = require("lodash/get");
+import { IAxis, IAxisNameConfig, IChartOptions } from "../../../../interfaces/Config";
+import { XAxisTitleOptions, YAxisTitleOptions } from "highcharts";
+
+type HighchartsAxisTitle = XAxisTitleOptions | YAxisTitleOptions;
+
+function getHighchartsAxisTitleConfiguration(
+    chartOptions: IChartOptions,
+    axisPropsKey: string,
+): HighchartsAxisTitle {
+    const axisNameConfig: IAxisNameConfig = get(chartOptions, axisPropsKey.concat(".name"), {});
+    const title: HighchartsAxisTitle = {};
+
+    if (axisNameConfig.position) {
+        title.align = axisNameConfig.position; // low | middle | high
+    }
+
+    // config.visible should be true/undefined by default
+    if (axisNameConfig.visible === false) {
+        title.text = "";
+    }
+
+    return title;
+}
+
+export function getAxisNameConfiguration(chartOptions: IChartOptions) {
+    return {
+        xAxis: get(chartOptions, "xAxes", []).map((axis: IAxis) => {
+            if (!axis) {
+                return {};
+            }
+
+            const opposite = get(axis, "opposite", false);
+            const axisPropsKey = opposite ? "secondary_xAxisProps" : "xAxisProps";
+
+            return {
+                title: getHighchartsAxisTitleConfiguration(chartOptions, axisPropsKey),
+            };
+        }),
+        yAxis: get(chartOptions, "yAxes", []).map((axis: IAxis) => {
```

Comment:
do you know what is difference between 9 lines of code for getting nameConfiguration for xAxis and yAxis? Exactly 3 chars.
What about:
`const axisNameConfigGetter = (axisNamePrefix) => (chartOptions) => {
    get(chartOptions, `${axisNamePrefix}Axes`, []).map((axis: IAxis) => {
        if (!axis) {
            return {};
        }

        const opposite = get(axis, "opposite", false);
        const axisPropsKey = opposite ? `secondary_${axisNamePrefix}AxisProps` : `${axisNamePrefix}AxisProps`;

        return {
            title: getHighchartsAxisTitleConfiguration(chartOptions, axisPropsKey),
        };
    })
}`

and use it as 

`return {
    xAxis: axisNameConfigGetter('x')(chartOptions),
    yAxis: axisNameConfigGetter('y')(chartOptions),
}`

Reply:
Fixed